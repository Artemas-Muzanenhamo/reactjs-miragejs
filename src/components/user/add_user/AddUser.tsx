import React, {Component} from "react";
import {Field, Form, Formik} from "formik";


interface User {
    user: {
        name: string;
        surname: string;
    }
}

export class AddUser extends Component<any, User> {

    constructor(props: any, state: User) {
        super(props, state);
        this.state = {
            user: {
                name: '',
                surname: ''
            }
        };
    }

    private formTitle = "Add A User";
    private buttonValue = "Add User";

    render() {
        return <div>
            <Formik
                initialValues={{name: '', surname: ''}}
                onSubmit={async values => {
                    await new Promise(resolve => setTimeout(resolve, 500));
                    this.setState({
                        user: {
                                name: values.name,
                                surname: values.surname
                            }
                    })
                }}
            >
                <Form>
                    <fieldset className="uk-fieldset">
                        <legend className="uk-legend">{this.formTitle}</legend>

                        <div className="uk-margin">
                            <Field className="uk-input uk-form-width-large" type="text" placeholder="First Name"
                                   name="name"/>
                        </div>
                        <div className="uk-margin">
                            <Field className="uk-input uk-form-width-large" type="text" placeholder="Surname"
                                   name="surname"/>
                        </div>

                        <div>
                            <input className="uk-button uk-button-default" type="submit" value={this.buttonValue}/>
                        </div>
                    </fieldset>
                </Form>
            </Formik>
        </div>
    }
}
