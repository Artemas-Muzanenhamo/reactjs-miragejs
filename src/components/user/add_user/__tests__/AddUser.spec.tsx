import React from "react";
import renderer from "react-test-renderer";
import {AddUser} from "../AddUser";

describe('<AddUser /> component', () => {
    it('should render <AddUser /> component', () => {
        let addUserComponentJson = renderer.create(
            <AddUser/>
        ).toJSON();

        expect(addUserComponentJson).toMatchSnapshot();
    });
});
