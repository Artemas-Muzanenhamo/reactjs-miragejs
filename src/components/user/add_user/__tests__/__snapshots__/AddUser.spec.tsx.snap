// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddUser /> component should render <AddUser /> component 1`] = `
<div>
  <form
    action="#"
    onReset={[Function]}
    onSubmit={[Function]}
  >
    <fieldset
      className="uk-fieldset"
    >
      <legend
        className="uk-legend"
      >
        Add A User
      </legend>
      <div
        className="uk-margin"
      >
        <input
          className="uk-input uk-form-width-large"
          name="name"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="First Name"
          type="text"
          value=""
        />
      </div>
      <div
        className="uk-margin"
      >
        <input
          className="uk-input uk-form-width-large"
          name="surname"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Surname"
          type="text"
          value=""
        />
      </div>
      <div>
        <input
          className="uk-button uk-button-default"
          type="submit"
          value="Add User"
        />
      </div>
    </fieldset>
  </form>
</div>
`;
